### PR TITLE
default apiserver using the current context server and set oauth port incase not the default 8443

### PR DESCRIPTION
**Pull request by rawlingsj**

### CODE REVIEW

**Comment by rawlingsj** [comment]
[merge]

File: cmds/deploy.go [hunk 1]
```diff
@@ -783,12 +797,20 @@ func processTemplate(tmpl *tapi.Template, ns string, domain string, apiserver st
 	}
 	p := template.NewProcessor(generators)
 
+	ip, port, err := net.SplitHostPort(apiserver)
```
Comment (jimmidyson) [inline cmds/deploy.go]: As above.

File: cmds/deploy.go [hunk 2]
```diff
@@ -256,12 +257,25 @@ func deploy(f *cmdutil.Factory, d DefaultFabric8Deployment) {
...
 
 	// extract the ip address from the URL
-	ip := strings.Split(cfg.Host, ":")[1]
-	ip = strings.Replace(ip, "/", "", 2)
+	u, err := url.Parse(cfg.Host)
+	if err != nil {
+		util.Fatalf("%s", err)
+	}
+
+	ip, _, err := net.SplitHostPort(u.Host)
```
Comment (rawlingsj) [inline cmds/deploy.go]: Ah yeah thanks, I'd missed that, when you say check the type of the error, is this good enough:
```
if err != nil && !strings.Contains(err.Error(),"missing port in address") {
   util.Fatalf("%s", err)
}
```

Comment (jimmidyson) [inline cmds/deploy.go]: Yeah that's all we can do sadly.

Comment (rawlingsj) [inline cmds/deploy.go]: ok, addressed in https://github.com/fabric8io/gofabric8/pull/203

**Pull request merged**